python/ray/util/iter.py

```diff
@@ -154,6 +154,9 @@ def __init__(self, actor_sets: List["_ActorSet"], name: str):
         self.actor_sets = actor_sets
         self.name = name
 
+        # keep explicit reference to parent iterator for repartition
+        self.parent_iterator = []
```

Comment:
Can we make the parent iterator a required argument in the constructor to avoid forgetting to set it?

Reply:
I didn't want to expose it in the API since it's still possible to create a ParallelIterator without calling `from_items`, `from_range` etc., right?

Reply:
And it would also show up in the documentation. Ideally the user shouldn't know this thing even exists